File: src/main/java/com/manson/fo76/repository/PriceCheckRepository.java
package com.manson.fo76.repository;

import org.springframework.context.annotation.Primary;
import org.springframework.stereotype.Repository;

@Primary
@Repository
public abstract class PriceCheckRepository implements BasePriceCheckRepository {

  public PriceCheckRepository() {
    System.out.println("PriceCheckRepository!==============================================================");
  }
}

File: src/main/java/com/manson/fo76/repository/PriceCheckRepository.kt
package com.manson.fo76.repository

import com.manson.fo76.domain.fed76.pricing.PriceCheckCacheItem
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.stereotype.Repository

@Repository
interface PriceCheckRepository : MongoRepository<PriceCheckCacheItem?, String?> {
    fun findByRequestId(requestId: String): PriceCheckCacheItem?
}